### ISSUE TITLE
Compilation is always performed (even if all artifacts are up to date)

### ISSUE THREAD

**Issue by barakman** [sc-forks/solidity-coverage#183]
This problem is also viable when executing `truffle test`.

However, with truffle there's a workaround, in the form of a one-time call to `truffle compile`, which generates all the artifacts.

If you add a `contracts_build_directory` value in the `truffle-config.js` file, then you can even tell truffle where to store the artifacts.

Ideally, it should be outside of your working directory, since the latter is typically under some sort of source control, and `truffle compile` updates the artifacts (adding date at the end of each file) even when no changes have been made in the contracts source code.

In any case, there seem to be a bit of a mess when using `solidity-coverage`.

First of all, the `contracts_build_directory` value is completely ignored.

Second, the alternative `dir` option (which should be added in the `solcover.js` file), doesn't seem to work coherently.

`solidity-coverage` indeed creates the artifacts under the requested directory, but then eventually tries to fetch them from the default directory `coverageEnv`.

In short, there are in fact two problems reported here:
1. Compilation is always performed
2. Artifacts directory is kind of inconclusive

**Comment by barakman** [comment]
Created the file `solcover.js` instead of `.solcover.js`.
